pkg/bitcoin/electrs/electrs.go

```diff
@@ -0,0 +1,204 @@
+package electrs
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ipfs/go-log"
+
+	"github.com/keep-network/keep-core/pkg/bitcoin"
+)
+
+const (
+	// DefaultRequestTimeout is a default timeout used for HTTP requests.
+	DefaultRequestTimeout = 5 * time.Second
+	// DefaultRetryTimeout is a default timeout used for retries.
+	DefaultRetryTimeout = 60 * time.Second
+)
+
+var logger = log.Logger("keep-electrs")
+
+// Config holds configurable properties.
+type Config struct {
+	URL            string
+	RequestTimeout time.Duration
+	RetryTimeout   time.Duration
+}
+
+// Connection is a handle for interactions with Electrum.
+type Connection struct {
+	url          string
+	client       httpClient
+	retryTimeout time.Duration
+}
+
+type httpClient interface {
+	Get(url string) (*http.Response, error)
+	Post(url string, contentType string, body io.Reader) (*http.Response, error)
+}
+
+// Connect initializes handle with provided Config.
+func Connect(config Config) (bitcoin.Chain, error) {
```

Comment:
`Connect` won't return an error even if the URL provided in the config is wrong.
We will only learn about the wrong URL when we try to use the Bitcoin chain for the first time.
I would consider adding a connection test before we leave `Connect`, we did that such test when connecting to btc client in [the old relay maintainer](https://github.com/keep-network/tbtc/blob/d18ef9aec7656f0ec7d317ece3e3d5c7aca92cda/relay/pkg/btc/remote.go#L45)